<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path"          default="$(find cubicle_detect)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find cubicle_detect)/yolo_network_config/cfg"/>
  <!--<arg name="min_disparity"              default="8"/>-->
  <arg name="min_disparity"              default="7"/>
  <arg name="disparity_scope"            default="128"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="cubicle_detect" file="$(find cubicle_detect)/config/ros.yaml"/>
  <rosparam command="load" ns="cubicle_detect" file="$(find cubicle_detect)/config/yolov3_bdd.yaml"/>
  <!--<rosparam command="load" ns="cubicle_detect" file="$(find cubicle_detect)/config/yolov3-spp.yaml"/>-->


  <param name="use_sim_time" value="true" />
  <!--Rectify input stereo images  -->
  <include file="$(find undistort_images)/launch/stereo_long.launch" />

  <!-- Start darknet and ros wrapper -->
  <node pkg="cubicle_detect" type="cubicle_detect" name="cubicle_detect" output="screen" launch-prefix="$(arg launch_prefix)">
    <!--whether use one-channel as input-->
    <param name="use_grey"              value="false" />
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <param name="min_disparity"         value="$(arg min_disparity)" />
    <param name="disparity_scope"       value="$(arg disparity_scope)" />
    <param name="image_left_topic"      value="/long/left/image_rect_color" />
    <param name="image_right_topic"     value="/long/right/image_rect_color" />
    <param name="image_info_left"       value="/long/left/camera_info" />
    <param name="image_info_right"      value="/long/right/camera_info" />
    <param name="scale"                 value="2" />
    <!--<param name="scale"                 value="1" />-->
    <param name="enable_stereo"         value="true" />
    <param name="enable_classification" value="false" />
  </node>

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
